### PR TITLE
Add props_json functionality to zotonic 1.0

### PR DESCRIPTION
**Pull request by mmzeeman**
### Description

Adds the props_json functionality which was recently added to 0.x

### Checklist

- [ ] documentation updated
- [x] tests added
- [x] no BC breaks

### CODE REVIEW

**Comment by mworrell** [comment]
You might want to rebase - I have upgraded jsxrecord on the master branch.

**Comment by mmzeeman** [comment]
Yes, rebased. Can you review it also as it is kind of important code.

File: apps/zotonic_core/src/db/z_db.erl [hunk 1]
```diff
@@ -749,20 +827,32 @@ update_map_atom_arrays(Props) ->
...
+    P = case get_current_props(DbDriver, Connection, Table, Id, Context) of
+            {ok, OldProps} ->
+                UpdateProps#{<<"props_json">> => ?DB_PROPS_JSON( maps:merge(OldProps, NewProps) ) };
+            _ ->
+                UpdateProps#{<<"props_json">> => ?DB_PROPS_JSON( NewProps )}
+        end,
+
+    %% Clear the existing props column
+    case lists:member(<<"props">>, Cols) of
+        true -> P#{ <<"props">> => null };
```
Comment (mworrell) [inline apps/zotonic_core/src/db/z_db.erl:840]: Shouldn't that be `undefined` ?

**Review by mworrell — APPROVED** [review]

File: apps/zotonic_core/src/db/z_db.erl (hunk 1 above)
Comment (mmzeeman) [reply apps/zotonic_core/src/db/z_db.erl:840]: I thought that the old props will be removed when it is set to null.

**Pull request merged**